src/main/scala/org/virtuslab/yaml/YamlDecoder.scala

```diff
@@ -3,28 +3,17 @@ package org.virtuslab.yaml
 import scala.deriving._
 import scala.compiletime._
 
-import org.virtuslab.internal.load.parse.ParserImpl
-import org.virtuslab.internal.load.YamlReader
-import org.virtuslab.internal.load.StringYamlReader
-import org.virtuslab.internal.load.construct.Construct
-import org.virtuslab.internal.load.compose.Node
 import org.virtuslab.internal.YamlError
-import org.virtuslab.internal.load.compose.ComposerImpl
+import org.virtuslab.internal.load.{YamlReader, StringYamlReader}
+import org.virtuslab.internal.load.parse.ParserImpl
+import org.virtuslab.internal.load.construct.{Der}
+import org.virtuslab.internal.load.compose.{Node, ComposerImpl}
 
 trait YamlDecoder[T: Mirror.Of]:
```

Comment:
IMHO having `Mirror` constraints on type constructor isn't the best practice, since it disallows providing own (manual) implementations.